xarray/core/computation.py

```diff
@@ -1005,10 +1016,16 @@ def earth_mover_distance(first_samples,
         func = functools.partial(func, **kwargs)
 
     if vectorize:
-        if meta is None:
+        if (
+            dask == "parallelized"
+            and meta is None
+            and output_dtypes is not None
+            and isinstance(output_dtypes, list)
+        ):
             # set meta=np.ndarray by default for numpy vectorized functions
             # work around dask bug computing meta with vectorized functions: GH5642
-            meta = np.ndarray
+            # defer raising errors to _apply_blockwise (e.g. if output_dtypes is None)
+            meta = np.ndarray((0, 0), dtype=output_dtypes[0])
```

Comment:
shouldn't we still set `meta = np.ndarray` if no output dtype is specified?

Reply:
`output_dtypes` is required for `dask="parallelized"` and will error if it is missing:

https://github.com/pydata/xarray/blob/8051c47686ae23062f9008785563d62327735002/xarray/core/computation.py#L670-L674

so this wont take effect. I am also not very happy with my approach, but didn't want to copy the checks from `apply_blockwise` up here - suggestions?

Reply:
Maybe the cleaner workaround is to move this down in to `_apply_blockwise`? Would it be enough to pass `vectorize` down to that level and then set `meta` as you are doing here? 

Also, it seems like we should raise that error about `output_dtypes` only if `meta.dtype` has not been set?

Reply:
Yes, I agree it would be cleaner to thread `vectorize` through to `_apply_blockwise`.

> Also, it seems like we should raise that error about output_dtypes only if meta.dtype has not been set?

Depends how important `output_dtypes` is for `np.vectorize`.

I am happy to work more on this, but I think it would be good to discuss #4060 first, which might make this obsolete.